client/state/login/magic-login/actions.js

```diff
@@ -92,6 +89,20 @@ export const fetchMagicLoginRequestEmail = ( email, redirectTo ) => dispatch =>
 		} );
 };
 
+async function postMagicLoginRequest( url, bodyObj ) {
+	const response = await fetch( url, {
+		method: 'POST',
+		credentials: 'include',
+		headers: { Accept: 'application/json', 'Content-Type': 'application/x-www-form-urlencoded' },
+		body: stringify( bodyObj ),
+	} );
+
+	if ( response.ok ) {
+		return await response.json();
+	}
+	throw new HTTPError( response, await response.text() );
```

Comment:
I think this can still work incorrectly in the following scenario:
1. HTTP response headers arrive and say "404 Not found"
2. At that moment, `response` is created and the promise returned from `fetch` is resolved with it.
3. The response body can still be arriving: the HTTP request is not finished yet
4. `response.text()` is waiting for the full response body to arrive.
5. Socket can be closed before the response is fully transmitted, due to a network error.

At this moment, we have two errors: the 404 HTTP code in the `response` (resolved promise from `fetch`), and network error (rejected promise from `response.text()`.

Which one should we throw from `postMagicLoginRequest`? `HTTPError( 404 )`, or whatever is thrown from `response.text()`? With this code, we throw the second error.

It's a detail that doesn't matter for login, but if we wanted to write generic helpers for HTTP errors, we'd need to be careful.

Reply:
Thats's an interesting situation! We should probably account for it, however unlikely. At the moment, and considering the use-case, I think it's fine to leave it as-is. If we do end up expending `HTTPError` to other parts of the codebase we may want to reconsider, though.